### PR TITLE
docs: add tx filterer docs, fix minor mistakes

### PR DESCRIPTION
**Pull request by dimazhornyk**
# What :computer: 
* add tx filterer docs
* fix minor style mistakes/typos

# Why :hand:
* cover new feature with lacking docs

### CODE REVIEW

File: docs/build/developer-reference/l1-l2-interop.md [hunk 1]
```diff
@@ -25,6 +25,14 @@ L1 to L2 communication is governed by the [`IZkSync.sol`](https://github.com/mat
 - If you prefer to learn-by-doing, the [cross chain governance tutorial](../tutorials/smart-contract-development/cross-chain-tutorial.md) is a practical example of layer interoperability.
   :::
 
+#### L1 to L2 transaction filtering
+
+Hyperchain operator can filter L1→L2 transactions coming through the Diamond proxy. It is achieved by having an address
```
Comment (shahar4) [inline docs/build/developer-reference/l1-l2-interop.md:30]: nit: `Hyperchain operator` -> `A hyperchain operator` or `Hyperchain operators`

File: docs/zk-stack/components/zkEVM/bootloader.md [hunk 2]
```diff
@@ -49,8 +49,7 @@ We also ensure that transactions do not exceed the limits of the memory space al
 
 ## Transaction Types and Validation
 
-While the main transaction format is the internal []`Transaction`
-format](<https://github.com/code-423n4/2023-10-zksync/blob/ef99273a8fdb19f5912ca38ba46d6bd02071363d/code/system-contracts/contracts/libraries/TransactionHelper.sol#L25>),
+While the main transaction format is the internal [`Transaction` format](https://github.com/code-423n4/2023-10-zksync/blob/ef99273a8fdb19f5912ca38ba46d6bd02071363d/code/system-contracts/contracts/libraries/TransactionHelper.sol#L25),
```
Comment (shahar4) [inline docs/zk-stack/components/zkEVM/bootloader.md:52]: Nice! I think linking from the ever-public [era-contracts](https://github.com/matter-labs/era-contracts/blob/4aa7006153ad571643342dff22c16eaf4a70fdc1/system-contracts/contracts/libraries/TransactionHelper.sol#L25) repo is best.

File: docs/build/developer-reference/l1-l2-interop.md [hunk 3]
```diff
@@ -25,6 +25,14 @@ L1 to L2 communication is governed by the [`IZkSync.sol`](https://github.com/mat
...
+of the special contract called TransactionFilterer in the state transition storage. If the filterer exists, it is being
+called in the Mailbox facet with the tx details, and has to return whether the transaction can be executed or not. If
+you want to use this feature, you have to deploy the contract that implements `ITransactionFilterer` interface and set
+its address in the admin facet. Same setter can be used to remove the filterer, it has to be called with the `0` address.
```
Comment (shahar4) [inline docs/build/developer-reference/l1-l2-interop.md:34]: Can you please add links to the different contracts here?

File: docs/build/developer-reference/l1-l2-interop.md [hunk 4]
```diff
@@ -25,6 +25,14 @@ L1 to L2 communication is governed by the [`IZkSync.sol`](https://github.com/mat
 - If you prefer to learn-by-doing, the [cross chain governance tutorial](../tutorials/smart-contract-development/cross-chain-tutorial.md) is a practical example of layer interoperability.
   :::
 
+#### L1 to L2 transaction filtering
+
+Hyperchain operator can filter L1→L2 transactions coming through the Diamond proxy. It is achieved by having an address
+of the special contract called TransactionFilterer in the state transition storage. If the filterer exists, it is being
+called in the Mailbox facet with the tx details, and has to return whether the transaction can be executed or not. If
+you want to use this feature, you have to deploy the contract that implements `ITransactionFilterer` interface and set
```
Comment (shahar4) [inline docs/build/developer-reference/l1-l2-interop.md:33]: nit: `you have to deploy the contract that implements `ITransactionFilterer`` -> `you have to deploy a contract that implements the `ITransactionFilterer``

File: docs/build/developer-reference/l1-l2-interop.md [hunk 5]
```diff
@@ -25,6 +25,14 @@ L1 to L2 communication is governed by the [`IZkSync.sol`](https://github.com/mat
 - If you prefer to learn-by-doing, the [cross chain governance tutorial](../tutorials/smart-contract-development/cross-chain-tutorial.md) is a practical example of layer interoperability.
   :::
 
+#### L1 to L2 transaction filtering
+
+Hyperchain operator can filter L1→L2 transactions coming through the Diamond proxy. It is achieved by having an address
+of the special contract called TransactionFilterer in the state transition storage. If the filterer exists, it is being
```
Comment (shahar4) [inline docs/build/developer-reference/l1-l2-interop.md:31]: You don't mean that it has to be called that, right? The only reqs are to set its address in the right place and have it implement the right interface, right?

File: docs/build/developer-reference/l1-l2-interop.md (hunk 1 above)
Comment (dimazhornyk) [reply docs/build/developer-reference/l1-l2-interop.md]: done

File: docs/zk-stack/components/zkEVM/bootloader.md (hunk 2 above)
Comment (dimazhornyk) [reply docs/zk-stack/components/zkEVM/bootloader.md]: yep, fixed

File: docs/build/developer-reference/l1-l2-interop.md (hunk 3 above)
Comment (dimazhornyk) [reply docs/build/developer-reference/l1-l2-interop.md]: added

File: docs/build/developer-reference/l1-l2-interop.md (hunk 4 above)
Comment (dimazhornyk) [reply docs/build/developer-reference/l1-l2-interop.md]: done

File: docs/build/developer-reference/l1-l2-interop.md (hunk 5 above)
Comment (dimazhornyk) [reply docs/build/developer-reference/l1-l2-interop.md]: yes, that was a poor wording, fixed it

File: docs/build/developer-reference/l1-l2-interop.md [hunk 6]
```diff
@@ -25,6 +25,16 @@ L1 to L2 communication is governed by the [`IZkSync.sol`](https://github.com/mat
 - If you prefer to learn-by-doing, the [cross chain governance tutorial](../tutorials/smart-contract-development/cross-chain-tutorial.md) is a practical example of layer interoperability.
   :::
 
+#### L1 to L2 transaction filtering
+
+A hyperchain operator can filter L1→L2 transactions coming through the Diamond proxy. It is achieved by having an
+address of the special contract, responsible for transaction filtering, in the state transition storage. If the filterer
+exists, it is being called in the [Mailbox facet](https://github.com/matter-labs/era-contracts/blob/dev/l1-contracts/contracts/state-transition/chain-deps/facets/Mailbox.sol#L243)
```
Comment (MexicanAce) [inline docs/build/developer-reference/l1-l2-interop.md:32]: These links are for the `dev` branch. As these are not live, do we want to hold off on merging these changes until `1.5.0` is live?

Comment (dimazhornyk) [reply docs/build/developer-reference/l1-l2-interop.md:32]: Yeah, it makes sense to wait for `1.5.0` before merging. I'll keep track of it and ping you when it's ready to be merged

**Comment by itsacoyote** [comment]
@dimazhornyk if this is still valid, can we move these changes into a PR for the new repo at https://github.com/matter-labs/zksync-docs? 
We are closing PRs in this repo as it will be archived shortly.

**Comment by dimazhornyk** [comment]
> @dimazhornyk if this is still valid, can we move these changes into a PR for the new repo at https://github.com/matter-labs/zksync-docs? We are closing PRs in this repo as it will be archived shortly.

It's fine I moved it to the new one, just forgot to close this PR

**Pull request closed without merging**